### PR TITLE
api - polly retry

### PR DESCRIPTION
**Pull request by Gafs63**

### CODE REVIEW

**Review by brandicorel — APPROVED** [review]

File: src/AspNetCore.ApiClient/AspNetCore.ApiClient.csproj [hunk 1]
```diff
@@ -4,13 +4,14 @@
     <TargetFramework>netstandard2.1</TargetFramework>
     <Description>Provide http client extensions to request apis.</Description>
     <PackageTags>$(PackageTags);client;httpclient;aspnetcore;extensions</PackageTags>
-    <Version>1.0.0</Version>
+    <Version>1.0.1-beta1</Version>
```
Comment (KevinRecuerda) [inline src/AspNetCore.ApiClient/AspNetCore.ApiClient.csproj:7]: could be minor version here

File: src/AspNetCore.ApiClient/HttpClientFactoryExtensions.cs [hunk 2]
```diff
@@ -1,36 +1,49 @@
...
     using System.Net.Http;
     using Microsoft.Extensions.DependencyInjection;
+    using Polly;
     using RecShark.AspNetCore.ApiClient.Security;
     using RecShark.DependencyInjection;
     using Refit;
 
     public static class HttpClientFactoryExtensions
     {
+        public static ICollection<Func<PolicyBuilder<HttpResponseMessage>, IAsyncPolicy<HttpResponseMessage>>> DefaultPolicies =
```
Comment (KevinRecuerda) [inline src/AspNetCore.ApiClient/HttpClientFactoryExtensions.cs:15]: could use single object instead of collection

**Review by KevinRecuerda — CHANGES_REQUESTED** [review]

File: src/AspNetCore.ApiClient/HttpClientFactoryExtensions.cs (hunk 2 above)
Comment (KevinRecuerda) [reply src/AspNetCore.ApiClient/HttpClientFactoryExtensions.cs:15]: We can wrap inner func if necessary

File: src/AspNetCore.ApiClient/HttpClientFactoryExtensions.cs [hunk 3]
```diff
@@ -1,48 +1,64 @@
...
+                           .ConfigureClient(config, configure)
+                           .AddErrorPolicies(useDefaultPolicies);
         }
 
         public static IHttpClientBuilder AddRefitApiClient<T>(
             this IServiceCollection services,
             ApiClientConfig         config,
-            Action<HttpClient>      configure = null)
+            Action<HttpClient>      configure          = null,
+            bool                    useDefaultPolicies = true)
```
Comment (KevinRecuerda) [inline src/AspNetCore.ApiClient/HttpClientFactoryExtensions.cs:36]:
```suggestion
            bool                    useDefaultPolicy = true)
```

File: src/AspNetCore.ApiClient/HttpClientFactoryExtensions.cs [hunk 4]
```diff
@@ -1,48 +1,64 @@
...
         {
-            var injectableConfig = new ApiClientConfig<T>() { Value = config };
+            var injectableConfig = new ApiClientConfig<T>() {Value = config};
             services.AddSingleton<IApiClientConfig<T>>(injectableConfig);
 
             services.Load<SecurityModule>();
             return services;
         }
 
+        private static IHttpClientBuilder AddErrorPolicies(this IHttpClientBuilder builder, bool useDefaultPolicy)
```
Comment (KevinRecuerda) [inline src/AspNetCore.ApiClient/HttpClientFactoryExtensions.cs:55]:
```suggestion
        private static IHttpClientBuilder AddErrorPolicy(this IHttpClientBuilder builder, bool useDefaultPolicy)
```

File: src/AspNetCore.ApiClient/HttpClientFactoryExtensions.cs [hunk 5]
```diff
@@ -1,48 +1,64 @@
...
     using Microsoft.Extensions.DependencyInjection;
+    using Polly;
     using RecShark.AspNetCore.ApiClient.Security;
     using RecShark.DependencyInjection;
     using Refit;
 
     public static class HttpClientFactoryExtensions
     {
+        public static Func<PolicyBuilder<HttpResponseMessage>, IAsyncPolicy<HttpResponseMessage>> DefaultPolicy =
+            builder => builder.WaitAndRetryAsync(new[] {TimeSpan.FromMilliseconds(500)});
```
Comment (KevinRecuerda) [inline src/AspNetCore.ApiClient/HttpClientFactoryExtensions.cs:16]: maybe, also add circuit breaker as default

**Review by KevinRecuerda — CHANGES_REQUESTED** [review]

**Review by KevinRecuerda — APPROVED** [review]

**Pull request merged**